fb: cfb: Supporting multiple display

soburi opened the pull request:
This PR adds support for multi-display. This change affects APIs widely.
I also make some API improvements.
(Removal of font size restrictions, Function argument signedness changes, Remove needless arguments from font APIs)

- Introducing `struct cfb_display`
- Changed the argument `struct device *` to `struct cfb_framebuffer *`
- Rename APIs starting with `cfb_framebuffer_*`

Introduces struct cfb_display as a data structure to represent the display.
As a result, it is now possible to create multiple pieces of data managed
using a single static variable in previous implementations,
making it possible to handle multiple displays.
The old `struct cfb_framebuffer` remains but is now part of the
`struct cfb_display` data structure.
You can get cfb_framebuffer from cfb_display using the added API
`cfb_display_get_framebuffer`.

All places where a `struct device *` argument was passed to specify the
target will be replaced with a `struct cfb_framebuffer *`.
In line with this, API names starting with `cfb_framebuffer_*`,
which were redundant and inconsistent names, will remove
`_framebuffer` and change to `cfb_*` .

Related API and implementation improvement have also been committed.

* Remove unused arguments

`cfb_get_font_size`
`cfb_get_numof_fonts`

* Change coordinate specification to signed integer

`cfb_invert_area`
`cfb_print`
`cfb_draw_point` (cfb_position)
`cfb_draw_line`
`cfb_draw_rect`

* Removed the restriction that the vertical size of the font must be a multiple of 8.

---------------------

This PR is part of a series of CFB enhancements (second of four https://github.com/zephyrproject-rtos/zephyr/issues/72177). 
I would appreciate it if you could check the #67881  part first.

henrikbrixandersen commented on doc/releases/migration-guide-3.6.rst:
```diff
@@ -465,6 +465,64 @@ zcbor
   :c:func:`zcbor_log`, and :c:func:`zcbor_trace` with no parameters is gone, and in its place are
   :c:func:`zcbor_trace_file` and :c:func:`zcbor_trace`, both of which take a ``state`` parameter.
 
+CFB
+===
```
This needs to go into the migration guide for v3.7.0 instead.

soburi replied:
Sorry for my late response.
Update done.

soburi committed:
fb: cfb: Supporting multiple display

There are three major changes.

- Introducing `struct cfb_display`
- Changed the argument `struct device *` to `struct cfb_framebuffer *`
- Rename APIs starting with `cfb_framebuffer_*`

Introduces struct cfb_display as a data structure to represent the display.
As a result, it is now possible to create multiple pieces of data managed
using a single static variable in previous implementations,
making it possible to handle multiple displays.
The old `struct cfb_framebuffer` remains but is now part of the
`struct cfb_display` data structure.
You can get cfb_framebuffer from cfb_display using the added API
`cfb_display_get_framebuffer`.

All places where a `struct device *` argument was passed to specify the
target will be replaced with a `struct cfb_framebuffer *`.
In line with this, API names starting with `cfb_framebuffer_*`,
which were redundant and inconsistent names, will remove
`_framebuffer` and change to `cfb_*` .

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

soburi committed:
fb
